packages/shift-next/src/lib/with-redux-store.js

```diff
@@ -39,7 +40,11 @@ export default (App) => {
       appContext.reduxStore = reduxStore
 
       if (!reduxStore.getState().menu.loaded) {
-        await reduxStore.dispatch(readMenu(Config.get().menuRequest))
+        let options = {}
+        if (appContext.req) {
+          options.postRequestHook = response => { setSurrogateKeys(response.headers, appContext.req, appContext.res) }
+        }
+        await reduxStore.dispatch(readMenu(Config.get().menuRequest, options))
```

Comment:
This is where the menu request gets made, and the postRequestHook gets passed through to the api-client